src/client/components/DataContext.js

```diff
@@ -4,26 +4,70 @@ const DataContext = React.createContext();
 
```

Comment:
I don't really understand why this context exists.
In some components you pull data from this context, in others they pull their own data with fetch.

If the aim is to fetch the same data less often there are better ways to solve that.
If the concern is hitting the database too much, put a small cache in the backend.

In general, I would not worry about that in this kind of app.